contracts/exchange-wrappers/Starkware.sol

```diff
@@ -35,6 +61,56 @@ contract Starkware is I_StarkwareContract {
         uint256 quantizedAmount
     ) external override
     {
+        uint256 amount = quantizedAmount;
+
+        address tokenAddress = extractContractAddress(assetType);
+        IERC20 token = IERC20(tokenAddress);
+        uint256 exchangeBalanceBefore = token.balanceOf(address(this));
+        bytes memory callData = abi.encodeWithSelector(
+            token.transferFrom.selector,
+            msg.sender,
+            address(this),
+            amount
+        );
+        safeTokenContractCall(tokenAddress, callData);
+
+        uint256 exchangeBalanceAfter = token.balanceOf(address(this));
+        require(exchangeBalanceAfter >= exchangeBalanceBefore, "OVERFLOW");
+        // NOLINTNEXTLINE(incorrect-equality): strict equality needed.
+        require(
+            exchangeBalanceAfter == exchangeBalanceBefore + amount,
+            "INCORRECT_AMOUNT_TRANSFERRED"
+        );
+    }
+
+    function extractContractAddress(
+        uint256 assetType
+    ) internal returns (address) {
+        return USDC_ADDRESS;
+    }
+
+    /*
+      Safe wrapper around ERC20/ERC721 calls.
+      This is required because many deployed ERC20 contracts don't return a value.
+      See https://github.com/ethereum/solidity/issues/4116.
+    */
+    function safeTokenContractCall(address tokenAddress, bytes memory callData) internal {
```

Comment:
Can you just use `SafeERC20` instead of this? Example here: https://github.com/dydxfoundation/governance-private/blob/master/contracts/treasury/Treasury.sol#L19

Reply:
I cant access governance :(

Reply:
Also, wasn't clear at the time but this contract is just a mock